front/pages/api/poke/workspaces/[wId]/data_sources/[name]/bot_enabled.ts

```diff
@@ -0,0 +1,139 @@
+  botEnabled: boolean;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<BotEnabledResponseBody | ReturnedAPIErrorType>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const user = await getUserFromSession(session);
+
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  if (!user.isDustSuperUser) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      const { wId } = req.query;
+      if (!wId || typeof wId !== "string") {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The request query is invalid, expects { workspaceId: string }.",
+          },
+        });
+      }
+
+      const { name } = req.query;
+      if (name !== "managed-slack") {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The request query is invalid, expects name to be managed-slack.",
+          },
+        });
+      }
+
+      if (!req.body || typeof req.body.botEnabled !== "boolean") {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "Missing required parameters. Required: botEnabled (boolean)",
+          },
+        });
+      }
+      const { botEnabled } = req.body;
+
+      const workspace = await Workspace.findOne({
+        where: {
+          sId: wId,
+        },
+      });
+
+      if (!workspace) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "workspace_not_found",
+            message: "Could not find the workspace.",
+          },
+        });
+      }
+
+      const dataSource = await DataSource.findOne({
+        where: {
+          workspaceId: workspace.id,
+          name,
+        },
+      });
+
+      if (!dataSource) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "data_source_not_found",
+            message: "Could not find the data source.",
+          },
+        });
+      }
+
+      const connectorRes = await ConnectorsAPI.setBotEnabled(
+        dataSource.connectorId || "",
+        botEnabled
+      );
+
+      if (connectorRes.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: `An error occurred while enabling the bot.`,
+          },
+        });
```

Comment:
```
   return apiError(req, res, {
          status_code: 500,
          api_error: {
            type: "internal_server_error",
            message: `An error occurred while enabling the bot.`,
          },
         connectorRes.error
        });
```

Reply:
As per Aric there’s a way to pass the actual error (not in message) no?

Reply:
Oh missed that
I indeed added it to message, checking your point now

Reply:
Ok got it fixing